Source/AwsCommonRuntimeKit/io/HostResolver.swift

```diff
@@ -4,98 +4,93 @@
 import AwsCCommon
 import AwsCIo
 
-public typealias HostResolvedContinuation = CheckedContinuation<[HostAddress], Error>
-
-public protocol HostResolver: AnyObject {
-    var rawValue: UnsafeMutablePointer<aws_host_resolver> { get set }
-    var config: UnsafeMutablePointer<aws_host_resolution_config> { get }
-    func resolve(host: String) async throws -> [HostAddress]
-}
-
-public final class DefaultHostResolver: HostResolver {
-    public var rawValue: UnsafeMutablePointer<aws_host_resolver>
-    public var config: UnsafeMutablePointer<aws_host_resolution_config>
+public class HostResolver {
+    let rawValue: UnsafeMutablePointer<aws_host_resolver>
+    let maxTTL: Int
     private let allocator: Allocator
 
-    public init(eventLoopGroup elg: EventLoopGroup,
-                maxHosts: Int,
-                maxTTL: Int,
-                allocator: Allocator = defaultAllocator,
-                shutdownCallback: ShutdownCallback? = nil) throws {
-        let shutdownCallbackCore = ShutdownCallbackCore(shutdownCallback)
+    public static func makeDefault(eventLoopGroup: EventLoopGroup,
+                                   maxHosts: Int,
+                                   maxTTL: Int,
+                                   allocator: Allocator = defaultAllocator,
+                                   shutdownCallback: ShutdownCallback? = nil) throws -> HostResolver {
```

Comment:
Nit: A more Swifty format

```suggestion
    public static func makeDefault(
        eventLoopGroup: EventLoopGroup,
        maxHosts: Int,
        maxTTL: Int,
        allocator: Allocator = defaultAllocator,
        shutdownCallback: ShutdownCallback? = nil
    ) throws -> HostResolver {
```

Reply:
Thanks, I will update it in a separate formatting PR.